bundler.js

```diff
@@ -91,7 +91,11 @@ export async function installBundler(bundlerVersionInput, lockFile, platform, ru
   } else {
     const gem = path.join(rubyPrefix, 'bin', 'gem')
     const bundlerVersionConstraint = /^\d+\.\d+\.\d+/.test(bundlerVersion) ? bundlerVersion : `~> ${bundlerVersion}`
-    await exec.exec(gem, ['install', 'bundler', '-v', bundlerVersionConstraint])
+    if ((bundlerVersionConstraint === '~> 2') && (common.floatVersion(rubyVersion) >= 2.3)) {
+      await exec.exec(gem, ['update', '--system', '--no-document'])
```

Comment:
I guess this is to avoid the error like https://github.com/ruby/setup-ruby/pull/253#issuecomment-1004061786 ?

I dislike updating RubyGems though, even more so for a new release, it can cause more things to break confusingly.

Reply:
See https://github.com/rubygems/rubygems/issues/5245.

The most recent Bundler version is a bit more tightly coupled to RubyGems than in the past.  Outside of Windows, there are warnings thrown about using an older version of RubyGems that doesn't support x, y, and z.  I'll see if I can find some examples.

Generally, I don't like doing anything that isn't necessary.

This is the one part of the PR that I added so it would work, but I hope there's a better solution.

**Off-topic:** I need some more coffee.  It's earlier here.